### PR TITLE
refactor(bin): use `resolveConfigFile.sync()`

### LINKED ISSUE #2726: Use resolveConfigFile.sync in the CLI
As of #2722 we now have a `resolveConfigFile.sync` function which we could use in the CLI to clean up a lot of the option merging code.

cc @ikatyang

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #2726

I'm not sure if there are more places should be cleaned up.

- [x] refactor: `resolveConfig`
- [x] refactor: `getOptionsForFile`
- [x] remove: `mapSeries`
- more?

### CODE REVIEW

File: bin/prettier.js [hunk 1]
```diff
@@ -483,9 +477,8 @@ function eachFilename(patterns, callback) {
       }
       // Use map series to ensure idempotency
       mapSeries(ignorer.filter(filePaths), filePath => {
```
Comment (azz) [inline bin/prettier.js]: We can delete `mapSeries`. Maybe:

```js
ignorer.filter(filePaths).forEach(filePath =>
  callback(filePath, getOptionsForFile(filePath))
);
```

**Comment by azz** [comment]
lgtm. 

@lydell you suggested we do a larger cleanup of the file, was there any part in particular?

I was thinking we could pull some of the "functions" into separate files, e.g. `cli-find-config-file.js`, `cli-help.js`, `cli-list-different.js`, etc. Though this may be overkill.

**Comment by lydell** [comment]
@azz If this PR does the simplifications you had in mind after we added the sync config resolution, then it’s all fine by me. 👍 

The issue I have with bin/prettier.js is the “global” (top-level) `argv` variable. Sometimes that’s used, sometimes we pass around a local `argv` between functions. That’s a bit confusing. I’d rather get rid of all the top-level code and have everything in functions, passing around `argv` (and whatever data they need).

Since the changes in this PR are related to `argv` I was thinking that maybe it’d come naturally to change more stuff while at it (one never knows). My suggestion to do a larger cleanup was more of a “If you feel like it, it’s OK to some larger refactoring – it won’t be rejected, because it’s needed” kind of comment.

If you feel like it, @ikatyang, you can do some more refactoring – either in this PR, or in a follow-up one. Or we’ll merge the improvements here as-is :)

**Comment by ikatyang** [comment]
I'll try it in another PR for separating files and make it "pure" as possible.

**Pull request merged**